### PR TITLE
[Finishes #164829220 users can share links on social media

### PR DESCRIPTION
**Pull request by vino-jasuba**
#### Description
Currently, users have no way to share articles on other social media sites. This feature adds functionality to enable users to do this.

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [ ] End to end
- [ ] Integration

#### Checklist:
- [x] My code follows the style guide for this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
[164829220](https://www.pivotaltracker.com/story/show/164829220)

### CODE REVIEW

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -32,6 +34,28 @@ def to_representation(self, instance):
...
+
+        if request:
+            article_link = parse.quote(request.build_absolute_uri(article_link))
+            title = parse.quote(title)
+            description = parse.quote(description)
+
+            links['facebook'] = "https://www.facebook.com/sharer/sharer.php?u={}".format(article_link)
+            links['google_plus'] = "https://plus.google.com/share?url={}".format(article_link)
+            links['twitter'] = "https://twitter.com/home?status={}".format(article_link)
+            links['email'] = "mailto:?&subject={}&body={}".format(title, description, article_link)
```
Comment (ogiste) [inline authors/apps/articles/serializers.py]: What does the email link do?

**Review by ogiste — COMMENTED** [review]
Fix the builds and look into the email link.

File: authors/apps/articles/serializers.py [hunk 2]
```diff
@@ -32,6 +34,28 @@ def to_representation(self, instance):
...
+
+        request = self.context.get('request')
+
+        if request:
+            article_link = parse.quote(request.build_absolute_uri(article_link))
+            title = parse.quote(title)
+            description = parse.quote(description)
+
+            links['facebook'] = "https://www.facebook.com/sharer/sharer.php?u={}".format(article_link)
+            links['google_plus'] = "https://plus.google.com/share?url={}".format(article_link)
```
Comment (ogiste) [inline authors/apps/articles/serializers.py]: Great work. Please remove the google plus link as google plus is officially shutting down personal and brand accounts.

File: authors/apps/articles/serializers.py (hunk 1 above)
Comment (vino-jasuba) [reply authors/apps/articles/serializers.py]: It's supposed to open the default mail client for the user so that they can share articles via email.

**Review by ogiste — APPROVED** [review]

**Review by bl4ck4ndbr0wn — APPROVED** [review]
Nice implementation.
Good work @vino-jasuba

File: authors/apps/authentication/tests/test_mail.py [hunk 3]
```diff
@@ -41,8 +41,9 @@ def test_can_verify_mail(self):
         """Tests if the verification link works"""
         res = self.sign_up_user()
         link = re.search(
-            r'http:\/\/[a-zA-Z0-9]+\/[a-z]+\/[a-z]+-[a-z]+\/[-0-9a-z]+\/[A-z]+[0-9]+\/',
+            r'http:\/\/[a-zA-Z0-9]+\/[a-z]+\/[a-z]+-[a-z]+\/[-0-9a-z]+\/\w+\/',
             mail.outbox[0].body)
+        print("first", mail.outbox[0].body)
```
Comment (oagutu) [inline authors/apps/authentication/tests/test_mail.py]: Omit the print statements in this file

**Review by oagutu — COMMENTED** [review]
@vino-jasuba This looks good, just make the suggested minor change

**Pull request merged**